src/accelerate/utils/modeling.py

```diff
@@ -1066,7 +1066,12 @@ def get_balanced_memory(
     )
     # The last device is left with max_memory just in case the buffer is not enough.
     for idx in gpus_idx_list[:-1]:
-        max_memory[idx] = min(max_memory[0] if low_zero and idx == 0 else per_gpu, max_memory[idx])
+        if idx == 0 and not low_zero and module_sizes["model.embed_tokens"] > per_gpu * 0.9:
+            max_memory[idx] = min(module_sizes["model.embed_tokens"] * 1.3, max_memory[idx])
+        elif idx == 1 and low_zero and module_sizes["model.embed_tokens"] > per_gpu * 0.9:
+            max_memory[idx] = min(module_sizes["model.embed_tokens"] * 1.3, max_memory[idx])
+        else:
+            max_memory[idx] = min(max_memory[0] if low_zero and idx == 0 else per_gpu, max_memory[idx])
```

Comment:
Not every model have their embedding layers as `model.embed_tokens`. Since this is specific to transformers, I don't think this should live in accelerate too. We can modify `max_memory` directly there as it is computed. We are also trying to tackle similar issues with this PR https://github.com/huggingface/accelerate/pull/3066#issuecomment-2417065301. 

Maybe a good solution would be to check if there is a module that is > per_gpu, return a message saying that the model is unbalanced which will lead to the whole model being put on only one device and propose to the user to use "sequential" mode instead ? Or we could also do as you suggested and modify `max_memory` with the largest module size.

Reply:
Thanks for the suggestion @SunMarc, I updated the code to use the largest leave module size instead of hardcoding the embed_tokens layer